mne/io/pick.py

```diff
@@ -784,24 +788,163 @@ def _pick_aux_channels(info, exclude='bads'):
                       ref_meg=False, exclude=exclude)
 
 
-def _pick_data_or_ica(info):
+def _pick_data_or_ica(info, exclude=()):
     """Pick only data or ICA channels."""
     if any(ch_name.startswith('ICA') for ch_name in info['ch_names']):
-        picks = pick_types(info, exclude=[], misc=True)
+        picks = pick_types(info, exclude=exclude, misc=True)
+    else:
+        picks = _pick_data_channels(info, exclude=exclude, with_ref_meg=True)
+    return picks
+
+
+def _picks_to_idx(info, picks, none='data', exclude='bads', allow_empty=False,
+                  with_ref_meg=True):
+    """Convert and check pick validity."""
+    #
+    # None -> all, data, or data_or_ica (ndarray of int)
+    #
+    orig_picks = picks
+    if isinstance(info, int):
+        n_chan = info
+    else:
+        n_chan = info['nchan']
+    assert n_chan >= 0
+
+    if picks is None:
+        if isinstance(info, int):  # special wrapper for no real info
+            picks = np.arange(n_chan)
+        else:
+            picks = none  # let _picks_str_to_idx handle it
```

Comment:
capital N in `None`

Reply:
Actually not here -- it's actually a `kwarg` and meant to be "what should be done when the user supplies `None`". I'll add a comment

Reply:
ok!